src/challenge/Challenge.tsx

```diff
@@ -85,6 +94,19 @@ class Challenge
     100
   );
 
+  additionalFilesLoadCallback(filename: string, contents: string) {
+    this.setState({
+      additionalFilesLoaded: {
+        ...this.state.additionalFilesLoaded,
+        [filename]: contents,
+      },
+    });
+  }
+
+  getAdditionalFiles() {
+    return this.state.additionalFilesLoaded;
```

Comment:
consider deleting the getter. The context can just use this.state.additionalFilesLoaded, which keeps the interface lighter.